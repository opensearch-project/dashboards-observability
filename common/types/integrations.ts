/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

export type IntegrationConnectionType = 's3' | 'index' | 'securityLake';
