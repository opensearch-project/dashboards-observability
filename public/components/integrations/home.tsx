/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React, { useState } from 'react';
import { HashRouter, Route, RouteComponentProps, Switch } from 'react-router-dom';
import { EuiGlobalToastList } from '@elastic/eui';
import { Toast } from '@elastic/eui/src/components/toast/global_toast_list';
import { Integration } from './components/integration';
import { TraceAnalyticsCoreDeps } from '../trace_analytics/home';
import { ChromeBreadcrumb } from '../../../../../src/core/public';
import { AvailableIntegrationOverviewPage } from './components/available_integration_overview_page';
import { AddedIntegrationOverviewPage } from './components/added_integration_overview_page';
import { AddedIntegration } from './components/added_integration';

export type AppAnalyticsCoreDeps = TraceAnalyticsCoreDeps;

interface HomeProps extends RouteComponentProps, AppAnalyticsCoreDeps {
  parentBreadcrumbs: ChromeBreadcrumb[];
}

export const Home = (props: HomeProps) => {
  const { http, chrome } = props;

  const commonProps = {
    http,
    chrome,
  };

  return (
    <div>
      <HashRouter>
        <Switch>
          <Route
            exact
            path={'/available'}
            render={() => <AvailableIntegrationOverviewPage {...commonProps} />}
          />
          <Route
            exact
            path={['/', '/installed']}
            render={() => <AddedIntegrationOverviewPage {...commonProps} />}
          />
          <Route
            exact
            path={'/installed/:id+'}
            render={(routerProps) => (
              <AddedIntegration
                integrationInstanceId={decodeURIComponent(routerProps.match.params.id)}
                {...commonProps}
              />
            )}
          />
          <Route
            exact
            path={'/available/:id+'}
            render={(routerProps) => (
              <Integration
                integrationTemplateId={decodeURIComponent(routerProps.match.params.id)}
                {...commonProps}
              />
            )}
          />
        </Switch>
      </HashRouter>
    </div>
  );
};
