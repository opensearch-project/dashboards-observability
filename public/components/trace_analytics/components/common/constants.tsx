/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

// Conversion factor for nanoseconds to milliseconds
export const NANOS_TO_MS = 1e6;

export const MILI_TO_SEC = 1000;

export const pieChartColors = [
  '#7492e7',
  '#c33d69',
  '#2ea597',
  '#8456ce',
  '#e07941',
  '#3759ce',
  '#ce567c',
  '#9469d6',
  '#4066df',
  '#da7596',
];

export interface Span {
  traceId: string;
  spanId: string;
  traceState: string;
  parentSpanId: string;
  name: string;
  kind: string;
  startTime: string;
  endTime: string;
  durationInNanos: number;
  serviceName: string;
  events: any[];
  links: any[];
  droppedAttributesCount: number;
  droppedEventsCount: number;
  droppedLinksCount: number;
  traceGroup: string;
  traceGroupFields: {
    endTime: string;
    durationInNanos: number;
    statusCode: number;
  };
  status: {
    code: number;
  };
  instrumentationLibrary: {
    name: string;
    version: string;
  };
}

export interface ParsedHit {
  _index: string;
  _id: string;
  _score: number;
  _source: Span;
  sort?: any[];
}
