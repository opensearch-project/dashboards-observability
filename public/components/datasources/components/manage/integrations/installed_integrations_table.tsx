/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import {
  EuiSmallButton,
  EuiCompressedFieldSearch,
  EuiFlexGroup,
  EuiFlexItem,
  EuiFlyout,
  EuiIcon,
  EuiInMemoryTable,
  EuiLink,
  EuiPanel,
  EuiSpacer,
  EuiText,
  EuiTitle,
} from '@elastic/eui';
import escapeRegExp from 'lodash/escapeRegExp';
import React, { CSSProperties, useEffect, useState } from 'react';
import { INTEGRATIONS_BASE } from '../../../../../../common/constants/shared';
import { DatasourceType } from '../../../../../../common/types/data_connections';
import { basePathLink } from '../../../../../../common/utils/shared';
import { coreRefs } from '../../../../../framework/core_refs';
import { IntegrationHealthBadge } from '../../../../integrations/components/added_integration';
import { AvailableIntegrationsList } from '../../../../integrations/components/available_integration_overview_page';
import { AvailableIntegrationsTable } from '../../../../integrations/components/available_integration_table';
import { SetupIntegrationForm } from '../../../../integrations/components/setup_integration';

interface IntegrationInstanceTableEntry {
  name: string;
  locator: {
    name: string;
    id: string;
  };
  status: string;
  assets: number;
}

const labelFromDataSourceType = (dsType: DatasourceType): string | null => {
  switch (dsType) {
    case 'S3GLUE':
      return 'S3 Glue';
    case 'PROMETHEUS':
      return null; // TODO Prometheus integrations not supported so no label available
    default:
      console.error(`Unknown Data Source Type: ${dsType}`);
      return null;
  }
};

const INSTALLED_INTEGRATIONS_COLUMNS = [
  {
    field: 'locator',
    name: 'Instance Name',
    render: (locator: { name: string; id: string }) => {
      return (
        <EuiLink
          data-test-subj={`${locator.name}IntegrationLink`}
          href={basePathLink(`/app/integrations#/installed/${locator.id}`)}
        >
          {locator.name}
        </EuiLink>
      );
    },
  },
  {
    field: 'status',
    name: 'Status',
    render: (status: string) => {
      return <IntegrationHealthBadge status={status} />;
    },
  },
  { field: 'assets', name: 'Asset Count' },
];

const instanceToTableEntry = (
  instance: IntegrationInstanceResult
): IntegrationInstanceTableEntry => {
  return {
    name: instance.name,
    locator: { name: instance.name, id: instance.id },
    status: instance.status,
    assets: (instance.assets ?? []).length,
  };
};

const AddIntegrationButton = ({
  toggleFlyout,
  fill,
}: {
  fill?: boolean;
  toggleFlyout: () => void;
}) => {
  return (
    <EuiSmallButton fill={fill} onClick={toggleFlyout}>
      Add Integrations
    </EuiSmallButton>
  );
};

const NoInstalledIntegrations = ({ toggleFlyout }: { toggleFlyout: () => void }) => {
  return (
    <EuiFlexGroup direction="column" alignItems="center" gutterSize="xs">
      <EuiFlexItem grow={false}>
        <EuiIcon type="iInCircle" glyphName="iInCircle" size="xxl" />
      </EuiFlexItem>
      <EuiFlexItem grow={false}>
        <EuiText textAlign="center">
          {/* Default margin is too wide -- compress it a bit */}
          <p style={{ 'margin-bottom': '8px' } as CSSProperties}>
            <b>There are no installed Integrations</b>
            <br />
            Add integrations to get started.
          </p>
        </EuiText>
      </EuiFlexItem>
      <EuiFlexItem grow={false}>
        <AddIntegrationButton toggleFlyout={toggleFlyout} />
      </EuiFlexItem>
    </EuiFlexGroup>
  );
};

export const InstallIntegrationFlyout = ({
  closeFlyout,
  datasourceType,
  datasourceName,
  refreshInstances,
}: {
  closeFlyout: () => void;
  datasourceType: DatasourceType;
  datasourceName: string;
  refreshInstances: () => void;
}) => {
  const [availableIntegrations, setAvailableIntegrations] = useState({
    hits: [],
  } as AvailableIntegrationsList);

  const [isInstalling, setIsInstalling] = useState(false);

  useEffect(() => {
    if (!coreRefs.http) {
      return;
    }
    coreRefs.http.get(`${INTEGRATIONS_BASE}/repository`).then((exists) => {
      setAvailableIntegrations(exists.data);
    });
  }, []);

  const s3FilteredIntegrations = {
    hits: availableIntegrations.hits.filter((config) =>
      config.labels?.includes(labelFromDataSourceType(datasourceType) ?? '')
    ),
  };

  const [installingIntegration, setInstallingIntegration] = useState<string | null>(null);
  const maybeCloseFlyout = () => {
    if (!isInstalling) {
      closeFlyout();
    }
  };

  return (
    <EuiFlyout onClose={maybeCloseFlyout} hideCloseButton={isInstalling}>
      {installingIntegration === null ? (
        <AvailableIntegrationsTable
          loading={false}
          data={s3FilteredIntegrations}
          isCardView={true}
          setInstallingIntegration={setInstallingIntegration}
        />
      ) : (
        <SetupIntegrationForm
          integration={installingIntegration}
          unsetIntegration={() => setInstallingIntegration(null)}
          renderType="flyout"
          forceConnection={
            datasourceType === 'S3GLUE'
              ? {
                  name: datasourceName,
                  type: 's3',
                }
              : undefined
          }
          setIsInstalling={(installing: boolean, success?: boolean) => {
            setIsInstalling(installing);
            if (success) {
              closeFlyout();
              refreshInstances();
            }
          }}
        />
      )}
    </EuiFlyout>
  );
};

export const InstalledIntegrationsTable = ({
  integrations,
  datasourceType,
  datasourceName,
  refreshInstances,
}: {
  integrations: IntegrationInstanceResult[];
  datasourceType: DatasourceType;
  datasourceName: string;
  refreshInstances: () => void;
}) => {
  const [query, setQuery] = useState('');
  const filteredIntegrations = integrations
    .map(instanceToTableEntry)
    .filter((i) => i.name.match(new RegExp(escapeRegExp(query), 'i')));

  const [showAvailableFlyout, setShowAvailableFlyout] = useState(false);
  const toggleFlyout = () => setShowAvailableFlyout((prev) => !prev);

  const integrationsTable = (
    <>
      <EuiTitle>
        <h2>Installed Integrations</h2>
      </EuiTitle>
      <EuiSpacer />
      <EuiFlexGroup>
        <EuiFlexItem>
          <EuiCompressedFieldSearch
            fullWidth
            placeholder="Search..."
            onChange={(queryEvent) => {
              setQuery(queryEvent.target.value);
            }}
          />
        </EuiFlexItem>
        <EuiFlexItem grow={false}>
          <AddIntegrationButton fill={true} toggleFlyout={toggleFlyout} />
        </EuiFlexItem>
      </EuiFlexGroup>
      <EuiSpacer />
      <EuiInMemoryTable items={filteredIntegrations} columns={INSTALLED_INTEGRATIONS_COLUMNS} />
    </>
  );

  return (
    <>
      <EuiSpacer />
      <EuiPanel>
        {integrations.length > 0 ? (
          integrationsTable
        ) : (
          <NoInstalledIntegrations toggleFlyout={toggleFlyout} />
        )}
      </EuiPanel>
      {showAvailableFlyout ? (
        <InstallIntegrationFlyout
          closeFlyout={() => setShowAvailableFlyout(false)}
          datasourceType={datasourceType}
          datasourceName={datasourceName}
          refreshInstances={refreshInstances}
        />
      ) : null}
    </>
  );
};
