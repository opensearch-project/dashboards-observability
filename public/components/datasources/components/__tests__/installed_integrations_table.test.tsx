/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import { InstalledIntegrationsTable } from '../manage/integrations/installed_integrations_table';
import { TEST_INTEGRATION_SEARCH_RESULTS } from '../../../../../test/constants';

describe('Installed Integrations Table test', () => {
  configure({ adapter: new Adapter() });

  it('Renders the installed integrations table', async () => {
    const wrapper = mount(
      <InstalledIntegrationsTable integrations={TEST_INTEGRATION_SEARCH_RESULTS} />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('Renders the empty installed integrations table', async () => {
    const wrapper = mount(<InstalledIntegrationsTable integrations={[]} />);

    expect(wrapper).toMatchSnapshot();
  });
});
