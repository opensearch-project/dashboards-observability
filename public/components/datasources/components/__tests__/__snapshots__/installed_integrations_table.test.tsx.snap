// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Installed Integrations Table test Renders the empty installed integrations table 1`] = `
<InstalledIntegrationsTable
  integrations={Array []}
>
  <EuiSpacer>
    <div
      className="euiSpacer euiSpacer--l"
    />
  </EuiSpacer>
  <EuiPanel>
    <div
      className="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow"
    >
      <NoInstalledIntegrations>
        <EuiFlexGroup
          alignItems="center"
          direction="column"
          gutterSize="xs"
        >
          <div
            className="euiFlexGroup euiFlexGroup--gutterExtraSmall euiFlexGroup--alignItemsCenter euiFlexGroup--directionColumn euiFlexGroup--responsive"
          >
            <EuiFlexItem
              grow={false}
            >
              <div
                className="euiFlexItem euiFlexItem--flexGrowZero"
              >
                <EuiIcon
                  glyphName="iInCircle"
                  size="xxl"
                  type="iInCircle"
                >
                  <EuiIconBeaker
                    aria-hidden={true}
                    className="euiIcon euiIcon--xxLarge euiIcon-isLoading"
                    focusable="false"
                    glyphName="iInCircle"
                    role="img"
                    style={null}
                  >
                    <svg
                      aria-hidden={true}
                      className="euiIcon euiIcon--xxLarge euiIcon-isLoading"
                      focusable="false"
                      glyphName="iInCircle"
                      height={16}
                      role="img"
                      style={null}
                      viewBox="0 0 16 16"
                      width={16}
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M5.277 10.088c.02.014.04.03.057.047.582.55 1.134.812 1.666.812.586 0 1.84-.293 3.713-.88L9 6.212V2H7v4.212l-1.723 3.876Zm-.438.987L3.539 14h8.922l-1.32-2.969C9.096 11.677 7.733 12 7 12c-.74 0-1.463-.315-2.161-.925ZM6 2H5V1h6v1h-1v4l3.375 7.594A1 1 0 0 1 12.461 15H3.54a1 1 0 0 1-.914-1.406L6 6V2Z"
                      />
                    </svg>
                  </EuiIconBeaker>
                </EuiIcon>
              </div>
            </EuiFlexItem>
            <EuiFlexItem
              grow={false}
            >
              <div
                className="euiFlexItem euiFlexItem--flexGrowZero"
              >
                <EuiText
                  textAlign="center"
                >
                  <div
                    className="euiText euiText--medium"
                  >
                    <EuiTextAlign
                      textAlign="center"
                    >
                      <div
                        className="euiTextAlign euiTextAlign--center"
                      >
                        <p
                          style={
                            Object {
                              "margin-bottom": "8px",
                            }
                          }
                        >
                          <b>
                            There are no installed Integrations
                          </b>
                          <br />
                          Add integrations to get started.
                        </p>
                      </div>
                    </EuiTextAlign>
                  </div>
                </EuiText>
              </div>
            </EuiFlexItem>
            <EuiFlexItem
              grow={false}
            >
              <div
                className="euiFlexItem euiFlexItem--flexGrowZero"
              >
                <EuiButton
                  href="/app/integrations#available"
                >
                  <EuiButtonDisplay
                    baseClassName="euiButton"
                    element="a"
                    href="/app/integrations#available"
                    isDisabled={false}
                    rel="noreferrer"
                  >
                    <a
                      className="euiButton euiButton--primary"
                      disabled={false}
                      href="/app/integrations#available"
                      rel="noreferrer"
                      style={
                        Object {
                          "minWidth": undefined,
                        }
                      }
                    >
                      <EuiButtonContent
                        className="euiButton__content"
                        iconSide="left"
                        textProps={
                          Object {
                            "className": "euiButton__text",
                          }
                        }
                      >
                        <span
                          className="euiButtonContent euiButton__content"
                        >
                          <span
                            className="euiButton__text"
                          >
                            Add Integrations
                          </span>
                        </span>
                      </EuiButtonContent>
                    </a>
                  </EuiButtonDisplay>
                </EuiButton>
              </div>
            </EuiFlexItem>
          </div>
        </EuiFlexGroup>
      </NoInstalledIntegrations>
    </div>
  </EuiPanel>
</InstalledIntegrationsTable>
`;

exports[`Installed Integrations Table test Renders the installed integrations table 1`] = `
<InstalledIntegrationsTable
  integrations={
    Array [
      Object {
        "assets": Array [
          Object {
            "assetId": "9506c132-f466-4ce3-a875-f187ddec587c",
            "assetType": "index-pattern",
            "description": "ss4o_logs_waf-aws_waf-sample-sample",
            "isDefaultAsset": false,
            "status": "available",
          },
          Object {
            "assetId": "7770e5be-6f10-4435-9773-021c6188bfe5",
            "assetType": "visualization",
            "description": "logs-waf-Top Client IPs",
            "isDefaultAsset": false,
            "status": "available",
          },
          Object {
            "assetId": "36f26341-22f0-49c5-9820-f787afb4090c",
            "assetType": "dashboard",
            "description": "logs-waf-dashboard",
            "isDefaultAsset": true,
            "status": "available",
          },
        ],
        "creationDate": "2024-03-12T16:23:18.053Z",
        "dataSource": "ss4o_logs_waf-aws_waf-sample-sample",
        "id": "d5b55c60-e08c-11ee-9c80-ff3b93498fea",
        "name": "aws_waf-sample",
        "status": "available",
        "templateName": "aws_waf",
      },
    ]
  }
>
  <EuiSpacer>
    <div
      className="euiSpacer euiSpacer--l"
    />
  </EuiSpacer>
  <EuiPanel>
    <div
      className="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow"
    >
      <EuiTitle>
        <h2
          className="euiTitle euiTitle--medium"
        >
          Installed Integrations
        </h2>
      </EuiTitle>
      <EuiSpacer>
        <div
          className="euiSpacer euiSpacer--l"
        />
      </EuiSpacer>
      <EuiFlexGroup>
        <div
          className="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
        >
          <EuiFlexItem>
            <div
              className="euiFlexItem"
            >
              <EuiFieldSearch
                compressed={false}
                fullWidth={true}
                incremental={false}
                isClearable={true}
                isLoading={false}
                onChange={[Function]}
                placeholder="Search..."
              >
                <EuiFormControlLayout
                  compressed={false}
                  fullWidth={true}
                  icon="search"
                  isLoading={false}
                >
                  <div
                    className="euiFormControlLayout euiFormControlLayout--fullWidth"
                  >
                    <div
                      className="euiFormControlLayout__childrenWrapper"
                    >
                      <EuiValidatableControl>
                        <input
                          className="euiFieldSearch euiFieldSearch--fullWidth"
                          onChange={[Function]}
                          onKeyUp={[Function]}
                          placeholder="Search..."
                          type="search"
                        />
                      </EuiValidatableControl>
                      <EuiFormControlLayoutIcons
                        compressed={false}
                        icon="search"
                        isLoading={false}
                      >
                        <div
                          className="euiFormControlLayoutIcons"
                        >
                          <EuiFormControlLayoutCustomIcon
                            size="m"
                            type="search"
                          >
                            <span
                              className="euiFormControlLayoutCustomIcon"
                            >
                              <EuiIcon
                                aria-hidden="true"
                                className="euiFormControlLayoutCustomIcon__icon"
                                size="m"
                                type="search"
                              >
                                <EuiIconBeaker
                                  aria-hidden={true}
                                  className="euiIcon euiIcon--medium euiIcon-isLoading euiFormControlLayoutCustomIcon__icon"
                                  focusable="false"
                                  role="img"
                                  style={null}
                                >
                                  <svg
                                    aria-hidden={true}
                                    className="euiIcon euiIcon--medium euiIcon-isLoading euiFormControlLayoutCustomIcon__icon"
                                    focusable="false"
                                    height={16}
                                    role="img"
                                    style={null}
                                    viewBox="0 0 16 16"
                                    width={16}
                                    xmlns="http://www.w3.org/2000/svg"
                                  >
                                    <path
                                      d="M5.277 10.088c.02.014.04.03.057.047.582.55 1.134.812 1.666.812.586 0 1.84-.293 3.713-.88L9 6.212V2H7v4.212l-1.723 3.876Zm-.438.987L3.539 14h8.922l-1.32-2.969C9.096 11.677 7.733 12 7 12c-.74 0-1.463-.315-2.161-.925ZM6 2H5V1h6v1h-1v4l3.375 7.594A1 1 0 0 1 12.461 15H3.54a1 1 0 0 1-.914-1.406L6 6V2Z"
                                    />
                                  </svg>
                                </EuiIconBeaker>
                              </EuiIcon>
                            </span>
                          </EuiFormControlLayoutCustomIcon>
                        </div>
                      </EuiFormControlLayoutIcons>
                    </div>
                  </div>
                </EuiFormControlLayout>
              </EuiFieldSearch>
            </div>
          </EuiFlexItem>
          <EuiFlexItem
            grow={false}
          >
            <div
              className="euiFlexItem euiFlexItem--flexGrowZero"
            >
              <EuiButton
                fill={true}
                href="/app/integrations#available"
              >
                <EuiButtonDisplay
                  baseClassName="euiButton"
                  element="a"
                  fill={true}
                  href="/app/integrations#available"
                  isDisabled={false}
                  rel="noreferrer"
                >
                  <a
                    className="euiButton euiButton--primary euiButton--fill"
                    disabled={false}
                    href="/app/integrations#available"
                    rel="noreferrer"
                    style={
                      Object {
                        "minWidth": undefined,
                      }
                    }
                  >
                    <EuiButtonContent
                      className="euiButton__content"
                      iconSide="left"
                      textProps={
                        Object {
                          "className": "euiButton__text",
                        }
                      }
                    >
                      <span
                        className="euiButtonContent euiButton__content"
                      >
                        <span
                          className="euiButton__text"
                        >
                          Add Integration
                        </span>
                      </span>
                    </EuiButtonContent>
                  </a>
                </EuiButtonDisplay>
              </EuiButton>
            </div>
          </EuiFlexItem>
        </div>
      </EuiFlexGroup>
      <EuiSpacer>
        <div
          className="euiSpacer euiSpacer--l"
        />
      </EuiSpacer>
      <EuiInMemoryTable
        columns={
          Array [
            Object {
              "field": "locator",
              "name": "Instance Name",
              "render": [Function],
            },
            Object {
              "field": "status",
              "name": "Status",
              "render": [Function],
            },
            Object {
              "field": "assets",
              "name": "Asset Count",
            },
          ]
        }
        items={
          Array [
            Object {
              "assets": 3,
              "locator": Object {
                "id": "d5b55c60-e08c-11ee-9c80-ff3b93498fea",
                "name": "aws_waf-sample",
              },
              "name": "aws_waf-sample",
              "status": "available",
            },
          ]
        }
        responsive={true}
        tableLayout="fixed"
      >
        <EuiBasicTable
          columns={
            Array [
              Object {
                "field": "locator",
                "name": "Instance Name",
                "render": [Function],
              },
              Object {
                "field": "status",
                "name": "Status",
                "render": [Function],
              },
              Object {
                "field": "assets",
                "name": "Asset Count",
              },
            ]
          }
          items={
            Array [
              Object {
                "assets": 3,
                "locator": Object {
                  "id": "d5b55c60-e08c-11ee-9c80-ff3b93498fea",
                  "name": "aws_waf-sample",
                },
                "name": "aws_waf-sample",
                "status": "available",
              },
            ]
          }
          noItemsMessage="No items found"
          onChange={[Function]}
          responsive={true}
          tableLayout="fixed"
        >
          <div
            className="euiBasicTable"
          >
            <div>
              <EuiTableHeaderMobile>
                <div
                  className="euiTableHeaderMobile"
                >
                  <EuiFlexGroup
                    alignItems="baseline"
                    justifyContent="spaceBetween"
                    responsive={false}
                  >
                    <div
                      className="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsBaseline euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow"
                    >
                      <EuiFlexItem
                        grow={false}
                      >
                        <div
                          className="euiFlexItem euiFlexItem--flexGrowZero"
                        />
                      </EuiFlexItem>
                      <EuiFlexItem
                        grow={false}
                      >
                        <div
                          className="euiFlexItem euiFlexItem--flexGrowZero"
                        />
                      </EuiFlexItem>
                    </div>
                  </EuiFlexGroup>
                </div>
              </EuiTableHeaderMobile>
              <EuiTable
                id="random_html_id"
                responsive={true}
                tableLayout="fixed"
              >
                <table
                  className="euiTable euiTable--responsive"
                  id="random_html_id"
                  tabIndex={-1}
                >
                  <EuiScreenReaderOnly>
                    <caption
                      className="euiScreenReaderOnly euiTableCaption"
                    >
                      <EuiDelayRender
                        delay={500}
                      />
                    </caption>
                  </EuiScreenReaderOnly>
                  <EuiTableHeader>
                    <thead>
                      <tr>
                        <EuiTableHeaderCell
                          align="left"
                          data-test-subj="tableHeaderCell_locator_0"
                          key="_data_h_locator_0"
                        >
                          <th
                            className="euiTableHeaderCell"
                            data-test-subj="tableHeaderCell_locator_0"
                            role="columnheader"
                            scope="col"
                            style={
                              Object {
                                "width": undefined,
                              }
                            }
                          >
                            <CellContents
                              className="euiTableCellContent"
                              showSortMsg={false}
                            >
                              <span
                                className="euiTableCellContent"
                              >
                                <EuiInnerText>
                                  <EuiI18n
                                    default="{innerText}; {description}"
                                    token="euiTableHeaderCell.titleTextWithDesc"
                                    values={
                                      Object {
                                        "description": undefined,
                                        "innerText": "Instance Name",
                                      }
                                    }
                                  >
                                    <span
                                      className="euiTableCellContent__text"
                                      title="Instance Name"
                                    >
                                      Instance Name
                                    </span>
                                  </EuiI18n>
                                </EuiInnerText>
                              </span>
                            </CellContents>
                          </th>
                        </EuiTableHeaderCell>
                        <EuiTableHeaderCell
                          align="left"
                          data-test-subj="tableHeaderCell_status_1"
                          key="_data_h_status_1"
                        >
                          <th
                            className="euiTableHeaderCell"
                            data-test-subj="tableHeaderCell_status_1"
                            role="columnheader"
                            scope="col"
                            style={
                              Object {
                                "width": undefined,
                              }
                            }
                          >
                            <CellContents
                              className="euiTableCellContent"
                              showSortMsg={false}
                            >
                              <span
                                className="euiTableCellContent"
                              >
                                <EuiInnerText>
                                  <EuiI18n
                                    default="{innerText}; {description}"
                                    token="euiTableHeaderCell.titleTextWithDesc"
                                    values={
                                      Object {
                                        "description": undefined,
                                        "innerText": "Status",
                                      }
                                    }
                                  >
                                    <span
                                      className="euiTableCellContent__text"
                                      title="Status"
                                    >
                                      Status
                                    </span>
                                  </EuiI18n>
                                </EuiInnerText>
                              </span>
                            </CellContents>
                          </th>
                        </EuiTableHeaderCell>
                        <EuiTableHeaderCell
                          align="left"
                          data-test-subj="tableHeaderCell_assets_2"
                          key="_data_h_assets_2"
                        >
                          <th
                            className="euiTableHeaderCell"
                            data-test-subj="tableHeaderCell_assets_2"
                            role="columnheader"
                            scope="col"
                            style={
                              Object {
                                "width": undefined,
                              }
                            }
                          >
                            <CellContents
                              className="euiTableCellContent"
                              showSortMsg={false}
                            >
                              <span
                                className="euiTableCellContent"
                              >
                                <EuiInnerText>
                                  <EuiI18n
                                    default="{innerText}; {description}"
                                    token="euiTableHeaderCell.titleTextWithDesc"
                                    values={
                                      Object {
                                        "description": undefined,
                                        "innerText": "Asset Count",
                                      }
                                    }
                                  >
                                    <span
                                      className="euiTableCellContent__text"
                                      title="Asset Count"
                                    >
                                      Asset Count
                                    </span>
                                  </EuiI18n>
                                </EuiInnerText>
                              </span>
                            </CellContents>
                          </th>
                        </EuiTableHeaderCell>
                      </tr>
                    </thead>
                  </EuiTableHeader>
                  <EuiTableBody
                    bodyRef={[Function]}
                  >
                    <tbody>
                      <EuiTableRow
                        isSelected={false}
                      >
                        <tr
                          className="euiTableRow"
                        >
                          <EuiTableRowCell
                            align="left"
                            key="_data_column_locator_0_0"
                            mobileOptions={
                              Object {
                                "header": "Instance Name",
                                "render": undefined,
                              }
                            }
                            setScopeRow={false}
                            textOnly={false}
                          >
                            <td
                              className="euiTableRowCell"
                              style={
                                Object {
                                  "width": undefined,
                                }
                              }
                            >
                              <div
                                className="euiTableRowCell__mobileHeader euiTableRowCell--hideForDesktop"
                              >
                                Instance Name
                              </div>
                              <div
                                className="euiTableCellContent euiTableCellContent--overflowingContent"
                              >
                                <EuiLink
                                  className=""
                                  data-test-subj="aws_waf-sampleIntegrationLink"
                                  href="/app/integrations#/installed/d5b55c60-e08c-11ee-9c80-ff3b93498fea"
                                  key=".0"
                                >
                                  <a
                                    className="euiLink euiLink--primary"
                                    data-test-subj="aws_waf-sampleIntegrationLink"
                                    href="/app/integrations#/installed/d5b55c60-e08c-11ee-9c80-ff3b93498fea"
                                    rel="noreferrer"
                                  >
                                    aws_waf-sample
                                  </a>
                                </EuiLink>
                              </div>
                            </td>
                          </EuiTableRowCell>
                          <EuiTableRowCell
                            align="left"
                            key="_data_column_status_0_1"
                            mobileOptions={
                              Object {
                                "header": "Status",
                                "render": undefined,
                              }
                            }
                            setScopeRow={false}
                            textOnly={false}
                          >
                            <td
                              className="euiTableRowCell"
                              style={
                                Object {
                                  "width": undefined,
                                }
                              }
                            >
                              <div
                                className="euiTableRowCell__mobileHeader euiTableRowCell--hideForDesktop"
                              >
                                Status
                              </div>
                              <div
                                className="euiTableCellContent euiTableCellContent--overflowingContent"
                              >
                                <IntegrationHealthBadge
                                  className=""
                                  key=".0"
                                  status="available"
                                >
                                  <EuiHealth
                                    color="success"
                                  >
                                    <div
                                      className="euiHealth euiHealth--textSizeS"
                                    >
                                      <EuiFlexGroup
                                        alignItems="center"
                                        gutterSize="xs"
                                        responsive={false}
                                      >
                                        <div
                                          className="euiFlexGroup euiFlexGroup--gutterExtraSmall euiFlexGroup--alignItemsCenter euiFlexGroup--directionRow"
                                        >
                                          <EuiFlexItem
                                            grow={false}
                                          >
                                            <div
                                              className="euiFlexItem euiFlexItem--flexGrowZero"
                                            >
                                              <EuiIcon
                                                color="success"
                                                type="dot"
                                              >
                                                <EuiIconBeaker
                                                  aria-hidden={true}
                                                  className="euiIcon euiIcon--medium euiIcon--success euiIcon-isLoading"
                                                  focusable="false"
                                                  role="img"
                                                  style={null}
                                                >
                                                  <svg
                                                    aria-hidden={true}
                                                    className="euiIcon euiIcon--medium euiIcon--success euiIcon-isLoading"
                                                    focusable="false"
                                                    height={16}
                                                    role="img"
                                                    style={null}
                                                    viewBox="0 0 16 16"
                                                    width={16}
                                                    xmlns="http://www.w3.org/2000/svg"
                                                  >
                                                    <path
                                                      d="M5.277 10.088c.02.014.04.03.057.047.582.55 1.134.812 1.666.812.586 0 1.84-.293 3.713-.88L9 6.212V2H7v4.212l-1.723 3.876Zm-.438.987L3.539 14h8.922l-1.32-2.969C9.096 11.677 7.733 12 7 12c-.74 0-1.463-.315-2.161-.925ZM6 2H5V1h6v1h-1v4l3.375 7.594A1 1 0 0 1 12.461 15H3.54a1 1 0 0 1-.914-1.406L6 6V2Z"
                                                    />
                                                  </svg>
                                                </EuiIconBeaker>
                                              </EuiIcon>
                                            </div>
                                          </EuiFlexItem>
                                          <EuiFlexItem
                                            grow={false}
                                          >
                                            <div
                                              className="euiFlexItem euiFlexItem--flexGrowZero"
                                            >
                                              Active
                                            </div>
                                          </EuiFlexItem>
                                        </div>
                                      </EuiFlexGroup>
                                    </div>
                                  </EuiHealth>
                                </IntegrationHealthBadge>
                              </div>
                            </td>
                          </EuiTableRowCell>
                          <EuiTableRowCell
                            align="left"
                            key="_data_column_assets_0_2"
                            mobileOptions={
                              Object {
                                "header": "Asset Count",
                                "render": undefined,
                              }
                            }
                            setScopeRow={false}
                            textOnly={true}
                          >
                            <td
                              className="euiTableRowCell"
                              style={
                                Object {
                                  "width": undefined,
                                }
                              }
                            >
                              <div
                                className="euiTableRowCell__mobileHeader euiTableRowCell--hideForDesktop"
                              >
                                Asset Count
                              </div>
                              <div
                                className="euiTableCellContent"
                              >
                                <span
                                  className="euiTableCellContent__text"
                                >
                                  3
                                </span>
                              </div>
                            </td>
                          </EuiTableRowCell>
                        </tr>
                      </EuiTableRow>
                    </tbody>
                  </EuiTableBody>
                </table>
              </EuiTable>
            </div>
          </div>
        </EuiBasicTable>
      </EuiInMemoryTable>
    </div>
  </EuiPanel>
</InstalledIntegrationsTable>
`;
